mohawk/tests.py

```diff
@@ -999,3 +1019,25 @@ def test_validate_bewit_with_unknown_credentials(self):
             'other_id': self.credentials,
         })
         check_bewit(url, credential_lookup=credential_lookup, now=1356420407 + 10)
+
+
+class TestPayloadHash(Base):
+    def test_hash_string(self):
+        payload = "hello world"
+        h = calculate_payload_hash(payload, 'sha256', 'application/json')
+        assert h == b'VKcQum9v7uK+x70J+4yWUPxT+cRsYCw8dxTvExGaPks='
+
+    def test_hash_binary(self):
+        payload = b"\x00\xffhello world\xff\x00"
+        h = calculate_payload_hash(payload, 'sha256', 'application/json')
+        assert h == b'Pad106+P9O4UzqCaQA2npbLlGgasj9YQJE9EWXIQrSw='
+
+    def test_hash_file(self):
+        payload = six.BytesIO(b"\x00\xffhello world\xff\x00")
+        h = calculate_payload_hash(payload, 'sha256', 'application/json')
+        assert h == b'Pad106+P9O4UzqCaQA2npbLlGgasj9YQJE9EWXIQrSw='
+
+    def test_hash_file_read_blocks(self):
+        payload = six.BytesIO(b"\x00\xffhello world\xff\x00")
+        h = calculate_payload_hash(payload, 'sha256', 'application/json', block_size=1)
+        assert h == b'Pad106+P9O4UzqCaQA2npbLlGgasj9YQJE9EWXIQrSw='
```

Comment:
Asserting the precise encoded / hashed output seems like it would make for a fragile test. Also, it's not important to test this logic.

I suggest just checking that reading with small block sizes is equal to reading in large block sizes. For example:

```python
hash1 = calculate_payload_hash(payload, algo, content_type, block_size=1)
hash2 = calculate_payload_hash(payload, algo, content_type, block_size=1024)
self.assertEqual(hash1, hash2)
```

Please also use `assertEqual()` because this test suite doesn't use `py.test` (although I guess it should).

Reply:
Writing the test for this raised an interesting issue. Should mohawk be responsible for preserving the read position of the file object? Or should the caller be responsible for re-seeking if needed?

Reply:
Because you have to `seek(0)` after the first assertion? That seems acceptable. So, yeah, I'd say the caller should be responsible for re-seeking if necessary. That is, `mohawk` should just read the file and forget about it. However, if you can think of a real life situation where that might be a problem, let me know.

Reply:
No, it's probably best to leave it. Not all file objects are seekable.

Reply:
I agree